ktor-http/common/src/io/ktor/http/ContentTypes.kt

```diff
@@ -60,7 +60,8 @@ public class ContentType private constructor(
             return false
         }
 
-        if (pattern.contentSubtype != "*" && !pattern.contentSubtype.equals(contentSubtype, ignoreCase = true)) {
+        if (pattern.contentSubtype != "*" && contentSubtype != "*"
+            && !pattern.contentSubtype.equals(contentSubtype, ignoreCase = true)) {
```

Comment:
`contentTypes.any { if (isWildcard) it.match(contentType) else contentType.match(it) }`
such a fix will not help in the following case:
> route with accept(ContentType.Any) and request with ContentType.Something.Any

can we just use a wildcard check in `ContentType.match()`? What do you think? please see my last commit



By the way, the `TestEngineMultipartTest.testNonMultipart` test crashes after the fix, but the problem doesn't seem to be directly related to ContentType. Am I missing something?

> Content-Type header is required for multipart processing

Reply:
The only concern I have is that the fix changes semantics of `match` method.
The current behavior is that `text/plain match text/* == true` but `text/* match text/plain == false`. Changing this is too disruptive for patch release.

However, I agree that naming of `ContentType.match` is confusing and should be changed. I've created a separate issue to address it:
- [KTOR-7720](https://youtrack.jetbrains.com/issue/KTOR-7720) ContentType.match name is confusing

Reply:
Let's make a private extension-function in `HttpMultiAcceptRouteSelector`:

```kotlin
private fun ContentType.isCompatibleWith(other: ContentType): Boolean = when {
    this.contentType == "*" || other.contentType == "*" -> true
    this.contentSubtype.startsWith("*") -> other.match(this)
    else -> this.match(other)
}
```

I've not tested this code, so it might not work.

Reply:
Let's make a private extension-function in `HttpMultiAcceptRouteSelector`:

```kotlin
private fun ContentType.isCompatibleWith(other: ContentType): Boolean = when {
    this.contentType == "*" || other.contentType == "*" -> true
    this.contentSubtype.startsWith("*") -> other.match(this)
    else -> this.match(other)
}
```

I've not tested this code, so it might not work.